src/ol_infrastructure/applications/mitopen/__main__.py

```diff
@@ -119,15 +138,67 @@
     "RESOURCE_EFFECTIVELY_STAR": {},
 }
 
+gh_workflow_s3_bucket_permissions = [
+    {
+        "Action": [
+            "s3:ListBucket*",
+        ],
+        "Effect": "Allow",
+        "Resource": [
+            f"arn:aws:s3:::{app_storage_bucket_name}",
+        ],
+    },
+    {
+        "Action": [
+            "s3:GetObject*",
+            "s3:PutObject",
+            "s3:PutObjectAcl",
+            "s3:DeleteObject",
+        ],
+        "Effect": "Allow",
+        "Resource": [
+            f"arn:aws:s3:::{app_storage_bucket_name}/frontend/*",
+        ],
+    },
+]
+
+gh_workflow_policy_document = {
+    "Version": IAM_POLICY_VERSION,
+    "Statement": gh_workflow_s3_bucket_permissions,
+}
+
+gh_workflow_iam_policy = iam.Policy(
+    f"ol_mitopen_gh_workflow_iam_permissions_{stack_info.env_suffix}",
+    name=f"ol-mitopen-gh-workflow-permissions-{stack_info.env_suffix}",
+    path=f"/ol-applications/mitopen/{stack_info.env_suffix}/",
+    policy=lint_iam_policy(
+        gh_workflow_policy_document, stringify=True, parliament_config=parliament_config
+    ),
+)
+
+# Just create a static user for now. Some day refactor to use
+# https://github.com/hashicorp/vault-action
+gh_workflow_user = iam.User(
+    f"ol_mitopen_gh_workflow_user_{stack_info.env_suffix}",
+    name=f"mitopen-gh-workflow-{stack_info.env_suffix}",
+    tags=aws_config.tags,
+)
+iam.PolicyAttachment(
+    f"ol_mitopen_gh_workflow_user_{stack_info.env_suffix}",
+    policy_arn=gh_workflow_iam_policy.arn,
+    users=[gh_workflow_user.name],
+)
+
```

Comment:
It should be possible to use the GitHub provider to populate the IAM credentials directly into the repository's secrets for GH actions.

Reply:
OHHH. NEAT. I will look into that.